consensus/consortium/common/contract.go

```diff
@@ -117,36 +123,80 @@ func NewContractIntegrator(config *chainParams.ChainConfig, backend bind.Contrac
 		chainId:             config.ChainID,
 		roninValidatorSetSC: roninValidatorSetSC,
 		slashIndicatorSC:    slashIndicatorSC,
-		profileSC:           profileSC,
 		finalityTrackingSC:  finalityTrackingSC,
-		stakingSC:           stakingSC,
-		signTxFn:            signTxFn,
-		signer:              types.LatestSignerForChainID(config.ChainID),
-		coinbase:            coinbase,
+
+		roninValidatorSetABI: roninValidatorSetABI,
+		profileABI:           profileABI,
+		stakingABI:           stakingABI,
+
+		chainConfig: config,
+		ethAPI:      ethAPI,
+
+		signTxFn: signTxFn,
+		signer:   types.LatestSignerForChainID(config.ChainID),
+		coinbase: coinbase,
 	}, nil
 }
 
 // GetBlockProducers retrieves block producer addresses
-func (c *ContractIntegrator) GetBlockProducers(blockNumber *big.Int) ([]common.Address, error) {
-	callOpts := bind.CallOpts{
-		BlockNumber: blockNumber,
+func (c *ContractIntegrator) GetBlockProducers(blockHash common.Hash, blockNumber *big.Int) ([]common.Address, error) {
+	blockNr := rpc.BlockNumberOrHashWithHash(blockHash, false)
+	if !c.chainConfig.IsTripp(blockNumber) {
```

Comment:
I prefer flipping the condition here to to be same as other check. Something like
```
if c.chainConfig.IsTripp() {...}
```